### ISSUE TITLE
Issues with building

### ISSUE THREAD

**Issue by winkstinks** [James-Livesey/Sinecraft#12]
When I use fxsdk build-fx, it returns this error:

make: *** No rule to make target 'all-fx'.  Stop.

It used to return an error specifically about CMake, so I redownloaded Sinecraft, but that changed the error to the above one. Any help helps.

**Comment by James-Livesey** [comment]
Weird... It's a while I've touched `fxsdk` but I do remember it was a bit of a pain to set up (I had to do it twice: once for my desktop and once for my laptop — both were a pain to get it working).

I'd recommend seeing if `fxsdk` works as expected for you first before then attempting to build Sinecraft — try [creating a new project in `fxsdk`](https://git.planet-casio.com/Lephenixnoir/fxsdk#using-the-fxsdk-command-line-tools) and compiling that, then see if you get the same error (if so, then it's probably worth reinstalling `fxsdk` from scratch).

...If you do want to cut to the chase and get a pre-compiled version of Sinecraft, I did make a pre-release by the way: https://github.com/James-Livesey/Sinecraft/releases/tag/0.1.0

Hopefully this should help you!

**Comment by winkstinks** [comment]
Yeah, so after I made this issue, I realized I should test if I can move stuff from my computer to my calculator... and I can't. I'm using FA-124, and I can't find out how to get files that aren't alpha mem, set up, and default onto the calculator. I know this is off-topic, but you're my only currently active source on how to do this kind of stuff; it's nowhere else on the internet.

**Comment by winkstinks** [comment]
Would you happen to have a suggestion on what to do or a different software?

**Comment by James-Livesey** [comment]
My calculator supports USB file transfer natively so I don't know too much about using the FA-124, but it looks like if you install [p7](https://git.planet-casio.com/cake/p7utils), then you can use `fxsdk send-fx` to send the file over. Looks like it's a reimplementation of the protocol rather than using Casio's own implementation, so it might work better for you if you can't get Casio's to work. Hopefully it works for you!

**Comment by winkstinks** [comment]
P7's having issues with the install... I have no clue where to go from here..

**Comment by James-Livesey** [comment]
Strange... I wouldn't really know where to go from here myself but the [Universal Casio Forum](https://community.casiocalc.org/) might be able to help with file transfer issues. Other than that, resetting your calculator might also help but it's a bit of a last resort.

One thing I do know is that Casio calculators essentially have two filesystems: main memory (which contains alpha memory, setup data etc.) and storage memory (which is a later addition and uses the Fugue FAT filesystem).

When I mount my calculator over USB, my file manager (just my main OS file manager, not a special program or anything) shows the folders for areas of main memory and then alongside them are my storage memory files and folders — so both memory areas are combined together. If you've got nothing in storage memory, then you probably only see the folders for main memory and nothing else. Not sure whether it's like this at your end. However, if you do have the same thing as me, just try copying the .g1a file into the root of your calculator's drive.

Hopefully that helps in some way. Are you seeing any specific errors when attempting to copy files over to the calculator or is it just that it doesn't look like there's a storage area? Also, what model of calculator do you have by the way?

**Comment by winkstinks** [comment]
I haven't quite had time to check the forum, but I'm gonna respond so you know I've seen the reply. There isn't any errors, its just that I cant find out where to put files I want to put on the calculator. I can't access the calc's memory using Windows btw. I have a Casio fx9750-gii.

**Comment by James-Livesey** [comment]
Interesting... Would you be able to take a screenshot of the software you use/where you are able to see the current calculator data (setup and alpha memory etc.)? Just wondering how similar it is to what I use.

**Comment by winkstinks** [comment]
The FA-124 side of the program I believe pulls from the file image, but as you can tell when I put files in there they don't show up in the program. I also cannot view the files themselves with this.
![Screenshot 2024-10-30 180403](https://github.com/user-attachments/assets/9668e727-f7e6-4e66-9d70-1021f3c0019c)
![Screenshot 2024-10-30 180513](https://github.com/user-attachments/assets/1dcdcce0-1869-468a-acd0-307845913bcd)

**Comment by James-Livesey** [comment]
Interesting... Yeah, I haven't used this program before but presumably you have to upload a file to the FA-124, then transfer to the calculator using the ⬅️ button, right? I'd recommend looking at the user guide for the FA-124 too as I think you might be able to find a PDF of it online 😅

**Comment by winkstinks** [comment]
Yeah, I can. I had other games I was gonna try out, and I put tetris onto it. But, i get a memory error when I try to run it. It could just be a game error tho, and I can only import .g1m or .g2m, so how can I convert sinecraft to .g1m?

**Comment by James-Livesey** [comment]
I've been having a look at the forums, and it _appears_ that you can transfer .g1a files: https://www.casiopeia.net/forum/viewtopic.php?f=20&t=1455&p=12719#p12719

It doesn't say much about compatibility though... [fxSDK](https://git.planet-casio.com/Lephenixnoir/fxsdk#compatibility-and-support) appears to support the G II models as well as the G III models (which is the one I have) even though the G II use the SuperH-3 processor and G III the SuperH-4. That seems to be why there's a difference between .g1m and .g1a, but I'm not entirely sure. I'm pretty sure the architecture is largely unchanged between them though and so only a single compiler is needed.

Did you actually manage to find a way to get the .g1a file at least onto the calculator, or did you experience errors trying to get the file onto the FA-124? If it's the latter, then hopefully the forum I linked at the start of this message should help for that. But if it's the former, my best guess is that it's a firmware compatibility issue — maybe it's worth trying to update your calculator to the latest firmware as Sinecraft (or fxSDK libraries) might be expecting some system calls to be present that don't exist in older firmware versions. I certainly did a firmware update on my G III shortly after I got it brand-new so I won't be surprised if there's an update available for the G II.

Let me know how you get on!

**Comment by winkstinks** [comment]
Dang, I forgot about this. I've had some stuff I've had to get to, so I'm just replying again to confirm I see this. I'll probably try again this weekend.
